api/include/opentelemetry/global/provider.h

```diff
@@ -30,15 +28,35 @@ class DefaultTracerProvider final : public TracerProvider
 class Provider
 {
 public:
-  static TracerProvider *GetTracerProvider() { return getPtrRef().get(); }
-  static void SetTracerProvider(TracerProvider *tp) { getPtrRef().reset(tp); }
+  static TracerProvider *GetTracerProvider()
```

Comment:
This doesn't look atomic.

What happens with racing calls to set/get, and also initial get/get?

Reply:
Good call, I added a lock around the get and set methods.